optimum/intel/neural_compressor/configuration.py

```diff
@@ -90,3 +93,201 @@ def _create_distillation_config(config: Union[Dict, DistillationConfig]):
+                - "rtn": weight quantization based round to nearest will be applied.
+                - "gptq": A new one-shot weight quantization method based on approximate second-order information,
+                    that is both highly-accurate and highly efficient. The weights of each column are updated based on
+                    the fixed-scale pseudo-quantization error and the inverse of the Hessian matrix calculated from the activations.
+                    The updated columns sharing the same scale may generate a new max/min value,
+                    so the scale needs to be saved for restoration.
+                - "autoround": AutoRound is an advanced weight-only quantization algorithm for low-bits LLM inference.
+                    It's tailored for a wide range of models and consistently delivers noticeable improvements,
+                    often significantly outperforming SignRound with the cost of more tuning time for quantization.
+    """
+
+    def __init__(
+        self,
+        bits: int = 4,
+        sym: bool = False,
+        num_samples: Optional[int] = None,
+        batch_size: Optional[int] = None,
+        seq_len: Optional[int] = None,
+        tokenizer: Optional[str] = None,
+        dataset: Optional[Union[str]] = None,
+        group_size: Optional[int] = None,
+        scale_dtype: Optional[str] = None,
+        compute_dtype: Optional[str] = None,
+        use_layer_wise: Optional[bool] = False,
+        quant_method: Union[str, INCQuantizationMethod] = INCQuantizationMethod.RTN,
+        **kwargs,
+    ):
+        super().__init__(bits=bits, sym=sym, num_samples=num_samples, batch_size=batch_size, seq_len=seq_len)
+        self.tokenizer = tokenizer
+        self.dataset = dataset
+        self.quant_method = INCQuantizationMethod(quant_method) if isinstance(quant_method, str) else quant_method
+        self.group_size = group_size or (-1 if bits == 8 else 128)
+        self.scale_dtype = scale_dtype
+        self.compute_dtype = compute_dtype
+        self.use_layer_wise = use_layer_wise
+
+        # GPTQ parameters
+        if self.quant_method == INCQuantizationMethod.GPTQ:
+            self.desc_act = kwargs.get("desc_act", False)
+            self.damp_percent = kwargs.get("damp_percent", 0.01)
+            self.block_size = kwargs.get("block_size ", 128)
+            self.static_groups = kwargs.get("static_groups", False)
+            self.true_sequential = kwargs.get("true_sequential", False)
+        # AutoRound parameters
+        if self.quant_method == INCQuantizationMethod.AutoRound:
+            self.lr_scheduler = kwargs.get("lr_scheduler", None)
+            self.lr = kwargs.get("lr", 0)
+            self.minmax_lr = kwargs.get("minmax_lr", None)
+            self.enable_quanted_input = kwargs.get("enable_quanted_input", True)
+            self.enable_minmax_tuning = kwargs.get("enable_minmax_tuning", True)
+            self.quant_lm_head = kwargs.get("quant_lm_head", True)
```

Comment:
this way the attributes will differ depending on the `quant_format`, what about having a `_quantization_parameters` dictionary instead ?